stat/stat_example_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat_test
+
+import (
+	"fmt"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/stat"
+)
+
+func ExampleLinearRegression() {
+
+	const (
+		n  = 100
+		p0 = 1
+		p1 = 3
+	)
+
+	xs := make([]float64, n)
+	ys := make([]float64, n)
+
+	fct := func(x float64) float64 {
+		return p0 + p1*x
+	}
+
+	for i := range xs {
+		xs[i] = float64(i)
+		ys[i] = fct(xs[i]) + 0.1*(rand.NormFloat64()-0.5)
```

Comment:
The `-0.5` just biases the random numbers to make the offset seem wrong. Remove it.